packages/navigator/src/browser/navigator-widget.tsx

```diff
@@ -100,6 +100,14 @@ export class FileNavigatorWidget extends FileTreeWidget {
     protected onActivateRequest(msg: Message): void {
         super.onActivateRequest(msg);
         this.selectionService.selection = this.model.selectedNodes;
+        const root = this.model.root;
+        if (CompositeTreeNode.is(root) && root.children.length === 1) {
+            const child = root.children[0];
+            if (SelectableTreeNode.is(child) && !child.selected && ExpandableTreeNode.is(child)) {
```

Comment:
What if something else is selected? It is not enough to check that `!child.selected`, we should check that the whole selection is empty.